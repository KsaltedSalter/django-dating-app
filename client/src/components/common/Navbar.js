import React from "react";
import { Link, useNavigate } from "react-router-dom";
//import { useParams } from "react-router";
// import { getSingleProfile } from "../../api/profileApi";
import { useParams } from "react-router";
// import { getPayload } from "../../api/authToken";
import logo from "../../styles/assets/whitelogo.png";
import { removeToken } from "../../api/authToken";

// import { getSingleUser } from "../../api/userApi";

const Navbar = () => {
  // const { id } = useParams();
  const navigate = useNavigate();
  const { id, user } = useParams();
  // const [account, setAccount] = useState({ account: null });

  // const getTheUser = async () => {
  //   try {
  //     const response = await getSingleUser(id);
  //     setAccount({ account: response.data });
  //     console.log("this is response.data", response.data);
  //   } catch (error) {
  //     console.error("An error occured getting single user", error);
  //   }
  // };

  // useEffect(() => {
  //   getTheUser();
  // }, []);

  // console.log("this is user", user);
  // console.log("this is account", account);

  // console.log("this is getPayload", getPayload().sub);

  // const isOwner = getPayload().sub === account.id;
  // console.log("isOwner is", isOwner);

  const handleClick = () => {
    removeToken();
    navigate("/");
  };

  return (
    <nav className="navbar navbar-expand-lg navbar-light">
      <div className="container-fluid">
        <span className="navbar-brand mb-0 h1">
          <img
            src={logo}
            alt=""
            width="40"
            height="35"
            className="d-inline-block align-text-top"
          />
          Ani-mate
        </span>
        <button
          className="navbar-toggler navbar-dark"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavAltMarkup"
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon navbar-dark"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav ms-auto">
            <Link to="/" className="nav-link">
              Home
            </Link>
            <Link to="/all" className="nav-link">
              Profiles
            </Link>
            <Link to={`/personas/myprofile/${user}`} className="nav-link">
              My Profile
            </Link>
            <Link to="/register" className="nav-link">
              Register
            </Link>
            <Link to="/login" className="nav-link">
              Login
            </Link>
            <button className="nav-link" onClick={handleClick}>
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
