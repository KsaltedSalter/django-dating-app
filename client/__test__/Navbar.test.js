import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";
import Navbar from "../src/components/common/Navbar";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => ({
    navigate: jest.fn()
  })
}));

describe("<Navbar />", () => {
  //   it("calls onNavChange when a button is clicked", () => {
  //     const mockOnNavChange = jest.fn();
  //     const component = render(
  //       <>
  //         <Router>
  //           <Navbar onNavChange={mockOnNavChange} />
  //         </Router>
  //       </>
  //     );

  //     const logoutButton = component.getByTestId("logout");
  //     fireEvent.click(logoutButton);

  //     expect(mockOnNavChange).toHaveBeenCalledTimes(1);
  //     expect(mockOnNavChange).toHaveBeenCalledWith("Logout");
  //   });

  //   it("has 6 items", () => {
  //     const { getByText } = render(
  //       <Router>
  //         <Navbar onNavChange={() => {}} />
  //       </Router>
  //     );
  //     expect(getByText("Home")).toBeInTheDocument();
  //     expect(getByText("Profiles")).toBeInTheDocument();
  //     expect(getByText("My Profile")).toBeInTheDocument();
  //     expect(getByText("Logout")).toBeInTheDocument();
  //     expect(getByText("Register")).toBeInTheDocument();
  //     expect(getByText("Login")).toBeInTheDocument();
  //   });

  it("matches the snapshot", () => {
    const mockOnNavChange = jest.fn();
    const { baseElement } = render(
      <Router>
        <Navbar onNavChange={mockOnNavChange} />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });

  //   it("matches snapshot on initial render", () => {
  //     const component = render(
  //       <Router>
  //         <Navbar />
  //       </Router>
  //     );
  //     expect(component.baseElement).toMatchSnapshot();
  //   });
});
