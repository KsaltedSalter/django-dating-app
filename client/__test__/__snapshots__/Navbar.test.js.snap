// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> matches the snapshot 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container-fluid"
      >
        <span
          class="navbar-brand mb-0 h1"
        >
          <img
            alt="ani-mate logo"
            class="d-inline-block align-text-top logo-img"
            height="35"
            src=""
            width="40"
          />
          Ani-mate
        </span>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler navbar-dark"
          data-bs-target="#navbarNavAltMarkup"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon navbar-dark"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            class="navbar-nav ms-auto"
          >
            <a
              class="nav-link"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              href="/register"
            >
              Register
            </a>
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
</body>
`;
